mlflow/tracking/_model_registry/client.py

```diff
@@ -214,6 +214,11 @@ def create_model_version(
                     )
                 mv = self.get_model_version(mv.name, mv.version)
                 sleep(AWAIT_MODEL_VERSION_CREATE_SLEEP_DURATION_SECONDS)
+            if mv.status != ModelVersionStatus.to_string(ModelVersionStatus.READY):
+                raise MlflowException(
+                    "Failed to successfully create the model version. \
+                Please check that the source location of the model is correct."
```

Comment:
I think we should give the model name, version, status like we do for the timeout above. Also, do we only expect the error to be related to source location? (probably, but I am not 100% sure) maybe better to display the status_message instead?